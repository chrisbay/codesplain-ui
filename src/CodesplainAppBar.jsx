import React from 'react';
import AppBar from 'material-ui/AppBar';

/**
 * A simple example of `AppBar` with an icon on the right.
 * By default, the left icon is a navigation-menu.
 */
const CodesplainAppBar = () => (
  <AppBar
    showMenuIconButton={false}
    title="Codesplain"
  />
);

export default CodesplainAppBar;
