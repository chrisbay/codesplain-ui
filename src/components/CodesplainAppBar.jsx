import React from 'react';
import AppBar from 'material-ui/AppBar';
import { browserHistory } from 'react-router';

const styles = {
  title: {
    cursor: 'pointer',
  }
}

/**
 * A simple example of `AppBar` with an icon on the right.
 * By default, the left icon is a navigation-menu.
 */
const CodesplainAppBar = () => (
  <AppBar
    showMenuIconButton={false}
    title="Codesplain"
    style={styles.title}
    onTitleTouchTap={() => { 
      browserHistory.push('/'); 
      location.reload();
    }}
  />
);

export default CodesplainAppBar;
