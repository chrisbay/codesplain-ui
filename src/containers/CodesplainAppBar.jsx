import React from 'react';
import { connect } from 'react-redux';
import AppBar from 'material-ui/AppBar';
import { browserHistory } from 'react-router';
import cookie from 'react-cookie';
import { saveUserSnippets, updateUserSnippets } from '../actions/app';

import LoginButton from '../components/buttons/LoginButton'
import AppMenu from '../components/menus/AppMenu';

const CLIENT_ID = process.env.REACT_APP_CLIENT_ID;
const GITHUB_URL = `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&scope=read:org`;

const styles = {
  title: {
    cursor: 'pointer',
  },
}

/*
<CodesplainAppBar /> renders as the banner located at the top of the screen. It
contains the text "Codesplain" on the far left, and either <AppMenu /> or
<LoginButton /> components on its right, depending on whether or not the user
has signed in with GitHub.
*/
class CodesplainAppBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: cookie.load('token') !== undefined,
    }
    this.handleSignOut = this.handleSignOut.bind(this);
    this.handleTitleClick = this.handleTitleClick.bind(this);
    this.onLoginClick = this.onLoginClick.bind(this);
    this.fetchUserSnippets = this.fetchUserSnippets.bind(this);
  }

  componentDidMount() {
    // const token = cookie.load('token');
    // const username = cookie.load('username');
    // this.fetchUserSnippets(token, username);
    const { dispatch } = this.props;
    dispatch(updateUserSnippets())
  }

  handleSignOut() {
    cookie.remove('token', { path: '/' });
    cookie.remove('username', { path: '/' });
    cookie.remove('userAvatarURL', { path: '/' });
    this.setState({ isLoggedIn: false });
    location.reload();
  }

  handleTitleClick(title) {
    console.log(`${title} was clicked`);
  }

  onLoginClick() {
    // Saves a cookie containing the current URL path, so we can retrieve it
    // later when GitHub sends the user back to the app, and send the user back
    // to the page they were on when they clicked the login button.
    const appState = JSON.stringify(this.props.appState);
    cookie.save('signInState', appState, { path: '/' });
    cookie.save('signInRedirect', location.pathname, { path: '/' });

    // Send the user to the GitHub OAuth authorization URL, where the client ID
    // is set to the CLIENT_ID env var.
    window.location = GITHUB_URL;
  }

  // Get meta data about the user's snippets and save to redux
  async fetchUserSnippets(token, username) {
    const { dispatch } = this.props;
    // let res;
    // try {
    //   const headers = {
    //     Accept: 'application/json',
    //     Authorization: `token ${token}`,
    //   };
    //   res = await axios.get(`${API_URL}/users/${username}/snippets`, { headers });
    // } catch(e) {
    //   return;
    // }
    const snippetMeta = {
      'test_snippet_1': {
        snippetName: 'Test Snippet 1',
        language:    'Python 3',
        lastEdited:  '2017-04-05T12:52:20.099Z',
        'private':   true,
      },
      'test_snippet_1': {
        snippetName: 'Test Snippet 2',
        language:    'Python 3',
        lastEdited:  '2017-04-05T12:52:20.099Z',
        'private':   true,
      }
    };

    dispatch(saveUserSnippets(snippetMeta));
  }

  render() {
    const { snippetMeta } = this.props.appState;
    const titles = snippetMeta ? 
      Object.entries(snippetMeta).map(snippet => snippet.snippetName)
      :
      null;
    const rightElement = this.state.isLoggedIn ?
      <AppMenu
        onSignOut={this.handleSignOut}
        snippetTitles={titles}
        onTitleClicked={this.handleTitleClick}
      />
      :
      <LoginButton
        onClick={this.onLoginClick}
      />;
    return (
      <AppBar
        showMenuIconButton={false}
        title="Codesplain"
        style={styles.title}
        onTitleTouchTap={() => {
          browserHistory.push('/');
          location.reload();
        }}
        iconElementRight={rightElement}
      />
    );
  }
}

const mapStateToProps = state => ({
  appState: state.app,
})

export default connect(mapStateToProps)(CodesplainAppBar);
