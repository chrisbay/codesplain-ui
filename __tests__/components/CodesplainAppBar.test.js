 import React from 'react'
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import CodesplainAppBar from '../../src/components/CodesplainAppBar'

describe('<CodesplainAppBar />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <CodesplainAppBar />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
