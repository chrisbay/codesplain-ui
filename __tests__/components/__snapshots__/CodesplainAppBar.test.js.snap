exports[`<CodesplainAppBar /> matches snapshot 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    Codesplain
  </h1>
</div>
`;
